extensions-contrib/scan-query/src/main/java/io/druid/query/scan/ScanQueryRunnerFactory.java

```diff
@@ -71,9 +71,7 @@ public ScanQueryRunnerFactory(
           final Query<ScanResultValue> query, final Map<String, Object> responseContext
       )
       {
-        final Number queryTimeout = query.getContextValue(QueryContextKeys.TIMEOUT, null);
-        final long timeoutAt = (queryTimeout == null || queryTimeout.longValue() == 0L)
-                               ? JodaUtils.MAX_INSTANT : System.currentTimeMillis() + queryTimeout.longValue();
+        final long timeoutAt = System.currentTimeMillis() + QueryContexts.getTimeout(query);
```

Comment:
Here timeoutAt is set to `System.currentTimeMillis()` if timeout is not specified, is that what you wanted to do?